test/core/ticktree/TickTreeUpdateOffer.t.sol

```diff
@@ -0,0 +1,347 @@
+  }
+
+  function test_update_offer_for_tick_MIN_TICK_where_higher_is_empty_and_lower_is_not_empty() public {
+    run_update_offer_scenarios_for_tick(MIN_TICK, emptyTickListSizeScenarios, singletonTickListSizeScenarios);
+  }
+
+  function test_update_offer_for_tick_MIN_TICK_where_higher_is_not_empty_and_lower_is_empty() public {
+    run_update_offer_scenarios_for_tick(MIN_TICK, singletonTickListSizeScenarios, emptyTickListSizeScenarios);
+  }
+
+  function test_update_offer_for_tick_MIN_TICK_where_higher_is_not_empty_and_lower_is_not_empty() public {
+    run_update_offer_scenarios_for_tick(MIN_TICK, singletonTickListSizeScenarios, singletonTickListSizeScenarios);
+  }
+
+  function run_update_offer_scenarios_for_tick(
+    int tick,
+    uint[] storage higherTickListSizeScenarios,
+    uint[] storage lowerTickListSizeScenarios
+  ) internal {
+    vm.pauseGasMetering();
+    TickScenario[] memory tickScenarios =
+      generateTickScenarios(tick, higherTickListSizeScenarios, lowerTickListSizeScenarios);
+    for (uint i = 0; i < tickScenarios.length; ++i) {
+      TickScenario memory tickScenario = tickScenarios[i];
+      for (uint j = 0; j < tickListScenarios.length; ++j) {
+        uint[2] storage tickListScenario = tickListScenarios[j];
+        run_update_offer_scenario(
+          UpdateOfferScenario({
+            tickScenario: tickScenario,
+            newTick: tickScenario.tick,
+            offerTickListSize: tickListScenario[0],
+            offerPos: tickListScenario[1]
+          }),
+          false
+        );
+        if (tickScenario.hasHigherTick) {
+          run_update_offer_scenario(
+            UpdateOfferScenario({
+              tickScenario: tickScenario,
+              newTick: tickScenario.higherTick,
+              offerTickListSize: tickListScenario[0],
+              offerPos: tickListScenario[1]
+            }),
+            false
+          );
+        }
+        if (tickScenario.hasLowerTick) {
+          run_update_offer_scenario(
+            UpdateOfferScenario({
+              tickScenario: tickScenario,
+              newTick: tickScenario.lowerTick,
+              offerTickListSize: tickListScenario[0],
+              offerPos: tickListScenario[1]
+            }),
+            false
+          );
+        }
+      }
+    }
+    vm.resumeGasMetering();
+  }
+
+  // function test_single_update_offer_scenario() public {
+  //   run_update_offer_scenario(
+  //     UpdateOfferScenario({
+  //       tickScenario: TickScenario({
+  //         tick: 0,
+  //         hasHigherTick: true,
+  //         higherTick: 524287,
+  //         higherTickListSize: 1,
+  //         hasLowerTick: true,
+  //         lowerTick: -16384,
+  //         lowerTickListSize: 0
+  //       }),
+  //       offerTickListSize: 1,
+  //       offerPos: 0,
+  //       newTick: -16384
+  //     }),
+  //     true
+  //   );
+  // }
```

Comment:
This can be uncommented now

Reply:
That specific scenario is already part of the `tick=0` tests :-) I've kept the commented out single-scenario tests as an easy way to run a single scenario.

But now that I think about it, it would ease maintenance if those tests are commented in. I'll do that!